receiver/awscontainerinsightreceiver/internal/host/ec2metadata.go

```diff
@@ -0,0 +1,96 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsec2metadata "github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	"go.uber.org/zap"
+)
+
+type Ec2metadata struct {
+	logger          *zap.Logger
+	refreshInterval time.Duration
+	instanceID      string
+	instanceType    string
+	shutdownC       chan bool
+}
+
+func NewEc2metadata(refreshInterval time.Duration, logger *zap.Logger) *Ec2metadata {
+	emd := &Ec2metadata{
+		refreshInterval: refreshInterval,
+		shutdownC:       make(chan bool),
+		logger:          logger,
+	}
+
+	emd.refresh()
+	go func() {
+		refreshTicker := time.NewTicker(emd.refreshInterval)
```

Comment:
I think this refresh and cancel logic is kind of duplicated (see it in the ebs volume one as well), maybe you can extract it as a util and reuse it.